packages/noya-state/src/reducers/layerReducer.ts

```diff
@@ -43,6 +48,122 @@ export function layerReducer(
         });
       });
     }
+    case 'groupLayer': {
+      const [, id, name] = action;
+
+      const ids = typeof id === 'string' ? [id] : id;
+
+      const page = getCurrentPage(state);
+      const pageIndex = getCurrentPageIndex(state);
+      const indexPaths = Layers.findAllIndexPaths(page, (layer) =>
+        ids.includes(layer.do_objectID),
+      );
+
+      const reversed = [...indexPaths.reverse()];
+      const selectedLayers: (Layers.ChildLayer | Sketch.Group)[] = [];
+
+      const addLayers = (layer: Sketch.AnyLayer) => {
+        if (Layers.isParentLayer(layer) && !Layers.isGroup(layer)) return;
```

Comment:
The `shapeGroup` layer type is a parent and can be included in groups (we don't support it yet though), so we may want to check specifically for the types to exclude. Page is impossible, so maybe `if (layer._class === 'artboard' || layer._class === 'symbolMaster') return`

This would rely on my suggestion above for `ChildLayer`